apply-ParagraphStandards.py

```diff
@@ -0,0 +1,149 @@
+import glob
+import json
+import yaml
+import getopt, sys
+from pprint import pprint
+from dateutil import parser
+
+def usage():
+    print ('Usage:')
+    print ("\t -m                               : modify source files")
+    print ("\t -t:{test}, or --test {test}      : run specific tests, or All (default)")
+    print ("\t -h, or --help                    : display help")
+    print ("\t Test list:")
+    print ("\t\t All")
+    print ("\t\t MDBlockGeneration")
+    print ("\t\t CodeBlockFix")
+
+try:
+    opts, args = getopt.getopt(sys.argv[1:], "-t:hm", ["help", "test=", "--modify"])
+except getopt.GetoptError as err:
+    # print help information and exit:
+    print(err) # will print something like "option -a not recognized"
+    usage()
+    sys.exit(2)
+modify = False
+tests = "All"
+for o, a in opts:
+    if o in ("-h", "--help"):
+        usage()
+        sys.exit()
+    elif o in ("-t", "--test"):
+        if a.lower() == 'all':
+            tests = 'All'
+        elif a.lower() == 'mdblockgeneration':
+            tests = 'MDBlockGeneration'
+        elif a.lower() == 'codeblockfix':
+            tests = 'CodeBlockFix'
+    elif o in ("-m", "--modify"):
+            modify = True
+    else:
+        assert False, "unhandled option"
+
+with open("./NotebookSanityExceptions.yaml", 'r') as stream:
+    sanity_exceptions = yaml.load(stream)
+
+notefiles = glob.glob('./**/note.json')
+
+for nf in notefiles:
+    has_changes = False
+    with open(nf, 'r+') as f:
+        notebook_id = nf.split('/')[1]
+        data = json.load(f)
+        # We will start at normal counting, so when you load the notebook,
+        # you can simply count to match with the reported paragraph number.
+        x = 1
+        for i in data['paragraphs']:
```

Comment:
How about using the name `paragraph` instead of `i` for improved readability, especially since `i` is typically used to represent an index (integer) value?